File: Java_MVVM/app/src/main/java/eu/gotoinc/requesinjava_mvvm/repository/ApiContractImpl.java
package eu.gotoinc.requesinjava_mvvm.repository;

import java.util.List;

import eu.gotoinc.requesinjava_mvvm.repository.data_models.responses.PaginationResponse;
import eu.gotoinc.requesinjava_mvvm.repository.data_models.responses.UserResponse;
import io.reactivex.Single;

public class ApiContractImpl implements ApiContract {
    private RetrofitRequest queries;

    public ApiContractImpl(RetrofitRequest queries) {
        this.queries = queries;
    }

    @Override
    public Single<PaginationResponse<List<UserResponse>>> getUsersList(int page) {
        return queries.getUsersList(page);
    }
}

File: Kotlin_MVVM/app/src/main/java/eu/gotoinc/kotlinmvvm/repository/ApiContractImpl.kt
package eu.gotoinc.kotlinmvvm.repository

import eu.gotoinc.kotlinmvvm.repository.datamodel.response.PaginationResponse
import eu.gotoinc.kotlinmvvm.repository.datamodel.response.UserResponse
import io.reactivex.Single

class ApiContractImpl(private val queries: RetrofitRequest) : ApiContract{

    override fun getUsersList(page: Int): Single<PaginationResponse<List<UserResponse>>> {
        return queries.getUsersList(page)
    }
}